src/document_save_as_text.lua

```diff
@@ -0,0 +1,377 @@
+function plugindef()
+    finaleplugin.RequireSelection = false
+    finaleplugin.NoStore = true
+    finaleplugin.Author = "Robert Patterson"
+    finaleplugin.Copyright = "CC0 https://creativecommons.org/publicdomain/zero/1.0/"
+    finaleplugin.Version = "1.0"
+    finaleplugin.Date = "May 30, 2022"
+    finaleplugin.CategoryTags = "Document"
+    finaleplugin.MinJWLuaVersion = 0.63
+    finaleplugin.Notes = [[
+        This script encodes the current document to a utf-8 text file. The primary purpose is to find changes
+        between one version of a document and another. One would then write each version out to a text file and then
+        use a comparison tool like kdiff3 to find differences. The text files could also be used to track changes with a tool like Git.
+
+        The specifics of the shorthand for how the music is represented may not be that important.
+        The idea is to identify the measures and staves that are different and then look at the score to see the differences.
+
+        The following are encoded in such a way that if they are different, a comparison tool will flag them.
+
+        - notes and rhythms
+        - articulations
+        - expressions (both text and shape)
+        - ties
+        - smart shapes
+        - lyric assignments
+
+        Chord symbols are currently not encoded, due to the lack of a simple way to generate a string for them. This is a needed
+        future enhancement.
+
+        The goal of this script is to assist in finding *substantive* differences that would affect how a player would play the piece.
+        The script encodes the items above but not small engraving differences such as placement coordinates. One hopes, for example,
+        that if there were a printed score that were out of date, this tool would flag the minimum number of changes that needed to
+        be hand-corrected in the older score.
+    ]]
+    return "Save Document As Text File...", "", "Write current document to text file."
+end
+
+local text_extension = ".txt"
+
+local note_entry = require('library.note_entry')
+local expression = require('library.expression')
+local enigma_string = require('library.enigma_string')
+local mixin = require('library.mixin')
+
+local fcstr = function(str)
+    local retval = finale.FCString()
+    retval.LuaString = str
+    return retval
+end
+
+function do_save_as_dialog(document)
+    local path_name = finale.FCString()
+    local file_name = finale.FCString()
+    local file_path = finale.FCString()
+    document:GetPath(file_path)
+    file_path:SplitToPathAndFile(path_name, file_name)
+    local full_file_name = file_name.LuaString
+    local extension = finale.FCString()
```

Comment:
Lines 58-60 could be shortened to the following, if you want:
`local extension = mixin.FCMString():SetLuaString(file_name.LuaString):ExtractFileExtension()`